### PR TITLE
Adds some events in the login process

### PR DESCRIPTION
**Pull request by shaoshiva**
Adds some events in the login process to be able to interact with it (eg. for double factor authentication).

Prevents the login popup to stack and allows popup chaining.

### CODE REVIEW

File: framework/classes/auth.php [hunk 1]
```diff
@@ -86,6 +113,11 @@ public static function check()
         // Only extend the cookie duration for the main request (preventing N 'Set-Cookie' headers if we have N HMVC requests)
         if (!\Request::is_hmvc()) {
 
+            // It's a new session (= auto-login using Cookie)
+            if (empty($session_user_id)) {
+                \Event::trigger('admin.beforeLoginSuccessWithCookie');
```
Comment (jguyomard) [inline framework/classes/auth.php]: a `trigger_function` with $logged_user_id seems better here

**Comment by Seb-C** [comment]
:+1:

File: framework/classes/auth.php (hunk 1 above)
Comment (shaoshiva) [inline framework/classes/auth.php]: Yes you're right, it should be a `trigger_function`.

**Comment by jguyomard** [comment]
:+1:

**Pull request merged**